So, my analysis: The Travis log shows that `TestMount` failed, so we'll just run that:
```
$ cd cmd/restic
$ go test -v -run TestMount
=== RUN   TestMount
Now serving the repository at /tmp/restic-test-153362927/mount
Don't forget to umount after quitting!
Now serving the repository at /tmp/restic-test-153362927/mount
Don't forget to umount after quitting!
Now serving the repository at /tmp/restic-test-153362927/mount
Don't forget to umount after quitting!
Now serving the repository at /tmp/restic-test-153362927/mount
Don't forget to umount after quitting!
--- FAIL: TestMount (1.39s)
        testing.go:30: using low-security KDF parameters for test
        testing.go:30: using low-security KDF parameters for test
        lock.go:77: setting lock timeout to 0s
        integration_test.go:51: repository initialized at /tmp/restic-test-153362927/repo
        integration_fuse_test.go:86: checking for 0 snapshots: []
        integration_fuse_test.go:47: mounted directory is ready
        integration_fuse_test.go:97: found 0 snapshots in repo: []
        integration_fuse_test.go:100: found 0 snapshots in fuse mount: []
        integration_fuse_test.go:113: Symlink latest isn't present in fuse dir
        integration_fuse_test.go:66: directory /tmp/restic-test-153362927/mount umounted
        integration_test.go:55: backing up [/tmp/restic-test-153362927/testdata]
        integration_fuse_test.go:86: checking for 1 snapshots: [06560e04]
        integration_fuse_test.go:47: mounted directory is ready
        integration_fuse_test.go:97: found 1 snapshots in repo: [06560e0412b58eac81cbf6dc9ddb960387178030eee191655f5d8030d223ff06]
        integration_fuse_test.go:100: found 2 snapshots in fuse mount: [2017-09-20T12:17:38+02:00 latest]
        integration_fuse_test.go:66: directory /tmp/restic-test-153362927/mount umounted
        integration_test.go:55: backing up [/tmp/restic-test-153362927/testdata]
        integration_fuse_test.go:86: checking for 2 snapshots: [06560e04 65a9af4a]
        integration_fuse_test.go:47: mounted directory is ready
        integration_fuse_test.go:97: found 2 snapshots in repo: [65a9af4a53a2827284e829e423ed2987d80cf9fe45954c1864ab9b10e2e95213 06560e0412b58eac81cbf6dc9ddb960387178030eee191655f5d8030d223ff06]
        integration_fuse_test.go:100: found 3 snapshots in fuse mount: [2017-09-20T12:17:38+02:00 2017-09-20T12:17:38+02:00-1 latest]
        integration_fuse_test.go:66: directory /tmp/restic-test-153362927/mount umounted
        integration_test.go:55: backing up [/tmp/restic-test-153362927/testdata]
        integration_fuse_test.go:86: checking for 3 snapshots: [06560e04 65a9af4a a75b6872]
        integration_fuse_test.go:47: mounted directory is ready
        integration_fuse_test.go:97: found 3 snapshots in repo: [a75b6872a9692af717f6b58bb3c5d6c3f640680b14f9d5e51f72f1ad682507bc 65a9af4a53a2827284e829e423ed2987d80cf9fe45954c1864ab9b10e2e95213 06560e0412b58eac81cbf6dc9ddb960387178030eee191655f5d8030d223ff06]
        integration_fuse_test.go:100: found 4 snapshots in fuse mount: [2017-09-20T12:17:38+02:00 2017-09-20T12:17:38+02:00-1 2017-09-20T12:17:38+02:00-2 latest]
        integration_fuse_test.go:66: directory /tmp/restic-test-153362927/mount umounted
FAIL
exit status 1
FAIL    github.com/restic/restic/cmd/restic     1.395s
```

Let's see what's going on here, the relevant error messages are:
```
--- FAIL: TestMount (1.39s)
        [...]
        integration_fuse_test.go:97: found 0 snapshots in repo: []
        integration_fuse_test.go:100: found 0 snapshots in fuse mount: []
        integration_fuse_test.go:113: Symlink latest isn't present in fuse dir
        [...]
```

So the repo is mounted, there are zero snapshots, and also the `latest` symlink isn't present. That's the error.

You need to change in `checkSnapshots()` so that it only checks for the presence of `latest` when there's more than zero snaphots.

--- a/cmd/restic/integration_fuse_test.go
+++ b/cmd/restic/integration_fuse_test.go
@@ -82,7 +82,7 @@ func listSnapshots(t testing.TB, dir string) []string {
 	return names
 }
 
-func checkSnapshots(t testing.TB, global GlobalOptions, repo *repository.Repository, mountpoint, repodir string, snapshotIDs restic.IDs) {
+func checkSnapshots(t testing.TB, global GlobalOptions, repo *repository.Repository, mountpoint, repodir string, snapshotIDs restic.IDs, expectedSnapshotsInFuseDir int) {
 	t.Logf("checking for %d snapshots: %v", len(snapshotIDs), snapshotIDs)
 
 	go testRunMount(t, global, mountpoint)
@@ -99,12 +99,22 @@ func checkSnapshots(t testing.TB, global GlobalOptions, repo *repository.Reposit
 	namesInSnapshots := listSnapshots(t, mountpoint)
 	t.Logf("found %v snapshots in fuse mount: %v", len(namesInSnapshots), namesInSnapshots)
 	Assert(t,
-		len(namesInSnapshots) == len(snapshotIDs),
-		"Invalid number of snapshots: expected %d, got %d", len(snapshotIDs), len(namesInSnapshots))
+		expectedSnapshotsInFuseDir == len(namesInSnapshots),
+		"Invalid number of snapshots: expected %d, got %d", expectedSnapshotsInFuseDir, len(namesInSnapshots))
 
 	namesMap := make(map[string]bool)
 	for _, name := range namesInSnapshots {
 		namesMap[name] = false
+	}
+
+	// Is "latest" present?
+	if len(namesMap) != 0 {
+		_, ok := namesMap["latest"]
+		if !ok {
+			t.Errorf("Symlink latest isn't present in fuse dir")
+		} else {
+			namesMap["latest"] = true
+		}
 	}
 
 	for _, id := range snapshotIDs {
@@ -153,7 +163,7 @@ func TestMount(t *testing.T) {
 	// We remove the mountpoint now to check that cmdMount creates it
 	RemoveAll(t, env.mountpoint)
 
-	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, []restic.ID{})
+	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, []restic.ID{}, 0)
 
 	SetupTarTestFixture(t, env.testdata, filepath.Join("testdata", "backup-data.tar.gz"))
 
@@ -163,7 +173,7 @@ func TestMount(t *testing.T) {
 	Assert(t, len(snapshotIDs) == 1,
 		"expected one snapshot, got %v", snapshotIDs)
 
-	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, snapshotIDs)
+	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, snapshotIDs, 2)
 
 	// second backup, implicit incremental
 	testRunBackup(t, []string{env.testdata}, BackupOptions{}, env.gopts)
@@ -171,7 +181,7 @@ func TestMount(t *testing.T) {
 	Assert(t, len(snapshotIDs) == 2,
 		"expected two snapshots, got %v", snapshotIDs)
 
-	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, snapshotIDs)
+	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, snapshotIDs, 3)
 
 	// third backup, explicit incremental
 	bopts := BackupOptions{Parent: snapshotIDs[0].String()}
@@ -180,7 +190,7 @@ func TestMount(t *testing.T) {
 	Assert(t, len(snapshotIDs) == 3,
 		"expected three snapshots, got %v", snapshotIDs)
 
-	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, snapshotIDs)
+	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, snapshotIDs, 4)
 }
 
 func TestMountSameTimestamps(t *testing.T) {
@@ -202,5 +212,5 @@ func TestMountSameTimestamps(t *testing.T) {
 		restic.TestParseID("5fd0d8b2ef0fa5d23e58f1e460188abb0f525c0f0c4af8365a1280c807a80a1b"),
 	}
 
-	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, ids)
+	checkSnapshots(t, env.gopts, repo, env.mountpoint, env.repo, ids, 4)
 }

--- a/internal/fuse/snapshots_dir.go
+++ b/internal/fuse/snapshots_dir.go
@@ -23,11 +23,13 @@ type SnapshotsDir struct {
 	root      *Root
 	snapshots restic.Snapshots
 	names     map[string]*restic.Snapshot
+	latest    string
 }
 
 // ensure that *SnapshotsDir implements these interfaces
 var _ = fs.HandleReadDirAller(&SnapshotsDir{})
 var _ = fs.NodeStringLookuper(&SnapshotsDir{})
+var _ = fs.NodeReadlinker(&snapshotLink{})
 
 // NewSnapshotsDir returns a new directory containing snapshots.
 func NewSnapshotsDir(root *Root, inode uint64, snapshots restic.Snapshots) *SnapshotsDir {
@@ -39,8 +41,16 @@ func NewSnapshotsDir(root *Root, inode uint64, snapshots restic.Snapshots) *Snap
 		names:     make(map[string]*restic.Snapshot, len(snapshots)),
 	}
 
+	// Track latest Snapshot
+	var latestTime time.Time
+	d.latest = ""
+
 	for _, sn := range snapshots {
 		name := sn.Time.Format(time.RFC3339)
+		if d.latest == "" || !sn.Time.Before(latestTime) {
+			latestTime = sn.Time
+			d.latest = name
+		}
 		for i := 1; ; i++ {
 			if _, ok := d.names[name]; !ok {
 				break
@@ -93,7 +103,47 @@ func (d *SnapshotsDir) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
 		})
 	}
 
+	// Latest
+	if d.latest != "" {
+		items = append(items, fuse.Dirent{
+			Inode: fs.GenerateDynamicInode(d.inode, "latest"),
+			Name:  "latest",
+			Type:  fuse.DT_Link,
+		})
+	}
 	return items, nil
+}
+
+type snapshotLink struct {
+	root     *Root
+	inode    uint64
+	target   string
+	snapshot *restic.Snapshot
+}
+
+func newSnapshotLink(ctx context.Context, root *Root, inode uint64, target string, snapshot *restic.Snapshot) (*snapshotLink, error) {
+	return &snapshotLink{root: root, inode: inode, target: target, snapshot: snapshot}, nil
+}
+
+func (l *snapshotLink) Readlink(ctx context.Context, req *fuse.ReadlinkRequest) (string, error) {
+	return l.target, nil
+}
+
+func (l *snapshotLink) Attr(ctx context.Context, a *fuse.Attr) error {
+	a.Inode = l.inode
+	a.Mode = os.ModeSymlink | 0777
+
+	if !l.root.cfg.OwnerIsRoot {
+		a.Uid = uint32(os.Getuid())
+		a.Gid = uint32(os.Getgid())
+	}
+	a.Atime = l.snapshot.Time
+	a.Ctime = l.snapshot.Time
+	a.Mtime = l.snapshot.Time
+
+	a.Nlink = 1
+
+	return nil
 }
 
 // Lookup returns a specific entry from the root node.
@@ -102,6 +152,16 @@ func (d *SnapshotsDir) Lookup(ctx context.Context, name string) (fs.Node, error)
 
 	sn, ok := d.names[name]
 	if !ok {
+		if name == "latest" && d.latest != "" {
+			sn2, ok2 := d.names[d.latest]
+
+			// internal error
+			if !ok2 {
+				return nil, fuse.ENOENT
+			}
+
+			return newSnapshotLink(ctx, d.root, fs.GenerateDynamicInode(d.inode, name), d.latest, sn2)
+		}
 		return nil, fuse.ENOENT
 	}
 

